packages/discord.js/typings/index.d.ts

```diff
@@ -1972,7 +1972,7 @@ export class AttachmentBuilder {
 
 export class Attachment {
   private constructor(data: APIAttachment);
-  public attachment: BufferResolvable | Stream;
+  private attachment: BufferResolvable | Stream;
```

Comment:
Considering it's initially set to `data.url`, which is a string:
```suggestion
  private attachment: BufferResolvable | Stream | string;
```